Sources/Swift/Integrations/SessionReplay/SentryOnDemandReplay.swift

```diff
@@ -21,40 +21,48 @@ enum SentryOnDemandReplayError: Error {
     case cantReadVideoSize
 }
 
-@available(iOS 16.0, tvOS 16.0, *)
 @objcMembers
 class SentryOnDemandReplay: NSObject {
     private let _outputPath: String
-    private let _onDemandDispatchQueue: DispatchQueue
-    
-    private var _starttime = Date()
-    private var _frames = [SentryReplayFrame]()
     private var _currentPixelBuffer: SentryPixelBuffer?
+    private var _totalFrames = 0
+    private let dateProvider: SentryCurrentDateProvider
+    private let workingQueue: SentryDispatchQueueWrapper
+    private var _frames = [SentryReplayFrame]()
+    
+    var frames: [SentryReplayFrame] {
+        _frames
```

Comment:
`m`: I think this still needs some synchorinization via the dispatch queue or an NSLock, because _frames could be modified on a background thread with the workingQueue, and this property could be called from any thread.